xbmc/pvr/windows/GUIWindowPVRGuide.cpp

```diff
@@ -207,71 +204,25 @@ void CGUIWindowPVRGuide::GetContextButtons(int itemNumber, CContextButtons &butt
 
 void CGUIWindowPVRGuide::UpdateSelectedItemPath()
 {
-  if (m_viewControl.GetCurrentControl() == GUIDE_VIEW_TIMELINE)
+  CGUIEPGGridContainer *epgGridContainer = (CGUIEPGGridContainer*) GetControl(m_viewControl.GetCurrentControl());
+  if (epgGridContainer)
   {
-    CGUIEPGGridContainer *epgGridContainer = (CGUIEPGGridContainer*) GetControl(m_viewControl.GetCurrentControl());
-    if (epgGridContainer)
-    {
-      CPVRChannelPtr channel(epgGridContainer->GetSelectedChannel());
-      if (channel)
-        SetSelectedItemPath(m_bRadio, channel->Path());
-    }
+    CPVRChannelPtr channel(epgGridContainer->GetSelectedChannel());
+    if (channel)
+      SetSelectedItemPath(m_bRadio, channel->Path());
   }
-  else
-    CGUIWindowPVRBase::UpdateSelectedItemPath();
 }
 
 void CGUIWindowPVRGuide::UpdateButtons(void)
 {
   CGUIWindowPVRBase::UpdateButtons();
-  switch (m_viewControl.GetCurrentControl())
-  {
-    case GUIDE_VIEW_TIMELINE: {
-      SET_CONTROL_LABEL(CONTROL_LABEL_HEADER1, g_localizeStrings.Get(19032));
-      break;
-    }
-    case GUIDE_VIEW_NOW: {
-      SET_CONTROL_LABEL(CONTROL_LABEL_HEADER1, g_localizeStrings.Get(19030));
-      break;
-    }
-    case GUIDE_VIEW_NEXT: {
-      SET_CONTROL_LABEL(CONTROL_LABEL_HEADER1, g_localizeStrings.Get(19031));
-      break;
-    }
-    case GUIDE_VIEW_CHANNEL: {
-      CPVRChannelPtr currentChannel(g_PVRManager.GetCurrentChannel());
-      if (currentChannel)
-        SET_CONTROL_LABEL(CONTROL_LABEL_HEADER1, currentChannel->ChannelName().c_str());
-      break;
-    }
-    default:
-      break;
-  }
-
+  SET_CONTROL_LABEL(CONTROL_LABEL_HEADER1, g_localizeStrings.Get(19032));
   SET_CONTROL_LABEL(CONTROL_LABEL_HEADER2, GetChannelGroup()->GroupName());
 }
 
 bool CGUIWindowPVRGuide::GetDirectory(const std::string &strDirectory, CFileItemList &items)
 {
-  switch (m_viewControl.GetCurrentControl())
-  {
-    case GUIDE_VIEW_TIMELINE:
-      GetViewTimelineItems(items);
-      break;
-    case GUIDE_VIEW_NOW:
-      GetViewNowItems(items);
-      break;
-    case GUIDE_VIEW_NEXT:
-      GetViewNextItems(items);
-      break;
-    case GUIDE_VIEW_CHANNEL:
-      GetViewChannelItems(items);
```

Comment:
Declaration and definition of this member function should be not used anymore and can be dropped as well.